app/src/main/java/woowacourse/shopping/data/dto/response/CouponDto.kt

```
@@ -0,0 +1,74 @@
+
+                "percentage" -> {
+                    requireNotNull(discount)
+                    val time = requireNotNull(availableTime)
+                    Discount.Percentage(
+                        discount,
+                        LocalTime.parse(time.start),
+                        LocalTime.parse(time.end),
+                    )
+                }
+
+                else -> throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)